mm2src/coins/rpc_command/account_balance.rs

```diff
@@ -0,0 +1,113 @@
+use crate::coin_balance::HDAddressBalance;
+use crate::hd_wallet::HDWalletCoinOps;
+use crate::rpc_command::hd_account_balance_rpc_error::HDAccountBalanceRpcError;
+use crate::{lp_coinfind_or_err, CoinBalance, CoinWithDerivationMethod, MmCoinEnum};
+use async_trait::async_trait;
+use common::mm_ctx::MmArc;
+use common::mm_error::prelude::*;
+use common::PagingOptionsEnum;
+use crypto::{Bip44Chain, RpcDerivationPath};
+use std::fmt;
+
+#[derive(Deserialize)]
+pub struct HDAccountBalanceRequest {
+    coin: String,
+    #[serde(flatten)]
+    params: AccountBalanceParams,
+}
+
+#[derive(Deserialize)]
+pub struct AccountBalanceParams {
+    pub account_index: u32,
+    pub chain: Bip44Chain,
+    #[serde(default = "common::ten")]
+    pub limit: usize,
+    #[serde(default)]
+    pub paging_options: PagingOptionsEnum<u32>,
+}
+
+#[derive(Debug, PartialEq, Serialize)]
+pub struct HDAccountBalanceResponse {
+    pub account_index: u32,
+    pub derivation_path: RpcDerivationPath,
+    pub addresses: Vec<HDAddressBalance>,
+    pub page_balance: CoinBalance,
+    pub limit: usize,
+    pub skipped: u32,
+    pub total: u32,
+    pub total_pages: usize,
+    pub paging_options: PagingOptionsEnum<u32>,
+}
+
+#[async_trait]
+pub trait AccountBalanceRpcOps {
+    async fn account_balance_rpc(
+        &self,
+        params: AccountBalanceParams,
+    ) -> MmResult<HDAccountBalanceResponse, HDAccountBalanceRpcError>;
+}
+
+pub async fn account_balance(
+    ctx: MmArc,
+    req: HDAccountBalanceRequest,
+) -> MmResult<HDAccountBalanceResponse, HDAccountBalanceRpcError> {
+    let coin = lp_coinfind_or_err(&ctx, &req.coin).await?;
+    match coin {
+        MmCoinEnum::UtxoCoin(utxo) => utxo.account_balance_rpc(req.params).await,
+        MmCoinEnum::QtumCoin(qtum) => qtum.account_balance_rpc(req.params).await,
+        _ => MmError::err(HDAccountBalanceRpcError::CoinIsActivatedNotWithHDWallet),
+    }
+}
+
+pub mod common_impl {
+    use super::*;
+    use crate::coin_balance::HDWalletBalanceOps;
+    use crate::hd_wallet::{HDAccountOps, HDWalletOps};
+    use common::calc_total_pages;
+
+    pub async fn account_balance_rpc<Coin>(
+        coin: &Coin,
+        params: AccountBalanceParams,
+    ) -> MmResult<HDAccountBalanceResponse, HDAccountBalanceRpcError>
+    where
+        Coin: HDWalletBalanceOps + CoinWithDerivationMethod<HDWallet = <Coin as HDWalletCoinOps>::HDWallet> + Sync,
+        <Coin as HDWalletCoinOps>::Address: fmt::Display + Clone,
+    {
+        let hd_wallet = coin.derivation_method().hd_wallet_or_err()?;
```

Comment:
No, I think it doesn't make the code more complicated. Will do :slightly_smiling_face:

Reply:
Refactored, but I left
```rust
let account_id = params.account_index;
```
variable, because it's used here
```diff
-            .or_mm_err(|| HDAccountBalanceRpcError::UnknownAccount { account_id })?;
+            .or_mm_err(|| HDAccountBalanceRpcError::UnknownAccount {
+                account_id: params.account_index,
+            })?;
```